constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -558,18 +557,48 @@ protected Request createSearchRequest(SearchRequest req, UserInfo userInfo) thro
 
             if (req.getCollectionId() != null) {
                 url = url.newBuilder()
-                .addQueryParameter("collection_id", req.getCollectionId())
-                .build();
+                    .addQueryParameter("collection_id", req.getCollectionId())
+                    .build();
             }
 
             if (req.getVariationsMap() != null) {
                 String variationsMapJson = new Gson().toJson(req.getVariationsMap());
                 url = url.newBuilder()
-                .addQueryParameter("variations_map", variationsMapJson)
-                .build();
+                    .addQueryParameter("variations_map", variationsMapJson)
+                    .build();
             }
 
-            Request request = this.makeUserRequestBuilder(userInfo)
+            if (req.getPreFilterExpression() != null) {
+                url = url.newBuilder()
+                    .addQueryParameter("pre_filter_expression", req.getPreFilterExpression())
+                    .build();
+            }
+
+            if (req.getQsParam() != null) {
+                url = url.newBuilder()
+                    .addQueryParameter("qs", req.getQsParam())
+                    .build();
+            }
+
+            if (req.getNow() != null) {
+                url = url.newBuilder()
+                    .addQueryParameter("now", req.getNow())
+                    .build();
+            }
+
+            if (req.getPage() != 1) {
+                url = url.newBuilder()
+                    .addQueryParameter("page", String.valueOf(req.getPage()))
+                    .build();
+            }
+
+            if (req.getOffset() != 0) {
+                url = url.newBuilder()
+                    .addQueryParameter("offset", String.valueOf(req.getOffset()))
+                    .build();
+            }
+
+            Request request = this.makeUserRequestBuilder(userInfo, true)
```

Comment:
Should this be "If now is passed, add auth. If not, do not" or do we always want to pass auth?

Reply:
Ooo, good point. Will update!